### ISSUE TITLE
Add setter for ionization potential

### ISSUE THREAD

**Issue by wtbarnes** [wtbarnes/fiasco#325]
When making an `Ion`, it is possible to set the abundance through a keyword argument. This keyword argument can be either the dataset name or an actual value. The ionization potential should work the same and should have similar setter/getter logic to how the abundance property works.

**Comment by jwreep** [comment]
Are there significantly different values or precisions from different atomic databases?  I guess I'm wondering why a setter is needed for ionization potential, as it should just be a fundamental property of an ion.

**Comment by wtbarnes** [comment]
That's a fair point. The only reason I proposed to do it this way is because the ionization potential data is structured very similarly to the ioneq and abundance data in the database. In practice, I think you're right that this data is unlikely to change much and not be very variable. There is only one file for these values in the database. The footer for this file does list several different sources:

```
filename: chianti.ip
ionization potentials: Martin, W.C., Sugar, J., Musgrove, A., Dalton, G.R.,
1995, NIST Database for Atomic Spectroscopy, Version 1.0, NIST Standard Reference Database 61
and Version 2.0 http://physics.nist.gov/cgi-bin/AtData/main_asd
ionization postentials: additional data for Cu and Zn from Biemeont, E., Fremat, Y., Quinet, P., 1999, ADNDT, 71, 117
ionization potentials: Erickson, G.W., 1977, J. Phys. Chem. Ref. Data, 6, 831
ionization potentials: C.E. Moore, 1970, NSRDS-NBS
ionization potentials: Boiko V.A., Pal'chikov V.G., Skobelev I.Yu., Faenov A.Ya., Boiko V.A., Pal'chikov V.G.,
Skobelev I.Yu., Faenov A.Ya. from - http://spectr-w3.snz.ru/index.phtml
produced as part of the Arcetri/Cambridge/NRL 'CHIANTI' atomic data base collaboration
K.P.Dere 2006
```

Whether these will ever be or need to be updated I do not know.

**Comment by jwreep** [comment]
Perhaps one reasonable thing would be to allow different files for input, so as to allow different sources for the values (CHIANTI, NIST, AtomDB, whatever).  I'm not sure it would make sense to allow the user to set the values manually, like `hydrogen[0].ionization_potential = 10*u.eV`.

**Comment by wtbarnes** [comment]
Yep that's the kind of use case I had in mind. With #342, this should now be possible.
